pkg/log/log_file.go

```diff
@@ -190,3 +199,66 @@ func checkLogFiles() {
 	checkFilesFn(warningLogFiles)
 	checkFilesFn(errorLogFiles)
 }
+
+// klogWriter is used in SetOutputBySeverity call below to redirect
+// any calls to klogv1 to end up in klogv2
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
+	if len(p) < defaultPrefixLength {
+		klog.InfoDepth(outputCallDepth, string(p))
+		return len(p), nil
+	}
+	if p[0] == 'I' {
+		klog.InfoDepth(outputCallDepth, string(p[defaultPrefixLength:]))
+	} else if p[0] == 'W' {
+		klog.WarningDepth(outputCallDepth, string(p[defaultPrefixLength:]))
+	} else if p[0] == 'E' {
+		klog.ErrorDepth(outputCallDepth, string(p[defaultPrefixLength:]))
+	} else if p[0] == 'F' {
+		klog.FatalDepth(outputCallDepth, string(p[defaultPrefixLength:]))
+	} else {
+		klog.InfoDepth(outputCallDepth, string(p[defaultPrefixLength:]))
+	}
+	return len(p), nil
+}
+
+// klogv1Redirect redirects klogv1 to klog.
+// Pulled from: https://github.com/kubernetes/klog/blob/master/examples/coexist_klog_v1_and_v2/coexist_klog_v1_and_v2.go
+// TODO: remove function once proxy is updated to use v2.
+func klogv1Redirect() {
+	var klogv1Flags flag.FlagSet
+	klogv1.InitFlags(&klogv1Flags)
+	klogv1Flags.Set("logtostderr", "false")
+	klogv1Flags.Set("stderrthreshold", "FATAL")
+	klogv1.SetOutputBySeverity("INFO", klogWriter{})
+}
+
+// init sets up klogv2 and redirects klogv1
+// "k8s.io/component-base/logs" is unusable
```

Comment:
should we add a TODO here as well to remove the code when / if "k8s.io/component-base/logs" is updated to use klogv2?